### PR TITLE
Added Block.addRunningEffects

### PR DESCRIPTION
**Pull request by ShetiPhian**
1.9 update of https://github.com/MinecraftForge/MinecraftForge/pull/2446

Adds Block.addRunningEffects to go with the existing ones enabling a mod to work around the vanilla particles.

### CODE REVIEW

File: patches/minecraft/net/minecraft/block/Block.java.patch [hunk 1]
```diff
@@ -834,6 +834,20 @@
...
 +    /**
++     * Allows a block to override the standard Entity.createRunningParticles particles.
++     * 
++     * @param state State at the specific world/pos
++     * @param worldObj The current world
++     * @param pos Position of the block that the entity is running on.
++     * @param entity The entity that is running on the block.
++     * @return True to prevent vanilla running particles form spawning.
++     */
++    public boolean addRunningEffects(IBlockState state, World worldObj, BlockPos pos, Entity entity)
```
Comment (kashike) [inline patches/minecraft/net/minecraft/block/Block.java.patch]: Don't name this `worldObj`, there's absolutely no reason to do so - instead, name it `world`.

Comment (ShetiPhian) [inline patches/minecraft/net/minecraft/block/Block.java.patch]: I was going to but notice the one above and bellow used worldObj and with them all being similar methods I left it. Most of the others have been cleaned up though so I'll change this as well.

File: patches/minecraft/net/minecraft/block/Block.java.patch [hunk 2]
```diff
@@ -834,6 +834,20 @@
 +    }
 +
 +    /**
++     * Allows a block to override the standard Entity.createRunningParticles particles.
++     * 
++     * @param state State at the specific world/pos
++     * @param worldObj The current world
```
Comment (iLexiconn) [inline patches/minecraft/net/minecraft/block/Block.java.patch]: Also update the javadoc

**Comment by darkhax** [comment]
Looks alright. This is actually something I plan to use as soon as it gets implemented.

**Comment by Actuarius** [comment]
@williewillus added labels [Feature]

**Pull request closed without merging**